admin/class-yoast-notification-center.php

```diff
@@ -735,6 +734,14 @@ private function retrieve_notifications_from_storage( $user_id ) {
 			// Apply array_values to ensure we get a 0-indexed array.
 			$notifications = array_values( array_filter( $notifications, [ $this, 'filter_notification_current_user' ] ) );
 
+			foreach ( $notifications as $notification ) {
+				$notification_has_been_updated = $notification->user_to_user_id();
+				// Just one notification changed is enough to update the storage.
+				if ( $notification_has_been_updated ) {
+					$this->notifications_need_storage = true;
+				}
+			}
+
```

Comment:
I detected an order problem. In `filter_notification_current_user` the `user_id` is already expected to be there.
But here you convert it only after the filter.

I was thinking of a solution:
* Move the whole loop above the filter
* Move the convert code into the `array_to_notification` method
But then I thought: why not check in the constructor of the notification? Specifically the `normalize_options` already has some logic that looks like a spot at first sight?